server/mediaserver.go

```diff
@@ -395,9 +413,27 @@ func gotRTMPStreamHandler(s *LivepeerServer) func(url *url.URL, rtmpStrm stream.
 							}
 							newSeg := &stream.HLSSegment{SeqNo: seg.SeqNo, Name: parseSegName(url), Data: data, Duration: seg.Duration}
 							s.LivepeerNode.VideoSource.InsertHLSSegment(sid, newSeg)
+
+							segHashes[i] = crypto.Keccak256(data)
 						}
-						for _, v := range res.Segments {
-							go dlFunc(v.Url)
+
+						for i, v := range res.Segments {
+							go dlFunc(v.Url, i)
+						}
+
+						cond.L.Lock()
+						for {
+							if n != 0 {
+								cond.Wait()
+							} else {
+								break
+							}
+						}
+						cond.L.Unlock()
+
+						if !eth.VerifySig(job.TranscoderAddress, crypto.Keccak256(segHashes...), res.Sig) {
```

Comment:
@j0sh I left `job` here as a placeholder... I was planning on grabbing it from "job" as it's defined on line 243, but now realize there are multiple `job`s per `bcasts`. Are we assuming that all these segments are coming from the same transcoder? Can we just grab the `job.TranscoderAddress` from one of these and include in the verification function on line 434?

Reply:
Here the `job` can be found within `rpcBcast.Job()` . Each `rpcBcast` is only active for one job at a time, and since we currently only allow one concurrent stream, that can be used safely.

When we do add multi stream support, we can have multiple `rpcBcast` instances.

Reply:
ohhh I see! That's the answer I was looking for. Thanks Josh!